src/playground/service.clj

```diff
@@ -25,15 +26,21 @@
 (defn all-invoices-page [request]
   (ring-resp/response (views/all-invoices (invoices.retrieve-all/perform request))))
 
-(defn invoice-page [request]
+(defn update-page [request]
   (let [user (invoices.retrieve/perform request)]
     (if user
-      (ring-resp/response (views/invoice user))
+      (ring-resp/response (views/update-invoice user))
       (ring-resp/not-found "Entry not in DB"))))
 
+(defn update-submit-page [request]
+  (ring-resp/response (views/submit-invoice (invoices.update/perform request))))
```

Comment:
This pattern is wrong. Handlers should be the responsible for invoking views, not the opposite.

Please fix it throughout this ns

Reply:
Do I then understand correctly that the pattern should be :

The route invoking the `perform` function that returns something like:
{:status 200 :body (views/foo)}

Reply:
Yes :)

> The route invoking the perform function 

We can call these 'handlers'